jax/_src/api.py

```diff
@@ -2364,37 +2406,18 @@ def _vjp_pullback_wrapper(cotangent_dtypes, cotangent_shapes,
   ans = fun(*args)
   return tree_unflatten(out_tree, ans)
 
+@overload
```

Comment:
Since we have access to `typing_extensions`, no need for this complicated if statement.

In Python 3.8+ we have access to `Literal` anyway.